### PR TITLE
Product search rebased

### PR DESCRIPTION
**Pull request by nirupa-kumar**
## Product Search (v1p3beta1)
This Pull Request adds the code samples for cloud.google.com documentation for Product Search How-to Guides

Product Search How-to Guides –> [vision/product-search/docs/how-to](https://cloud.google.com/vision/product-search/docs/how-to)

### products.js
- [x] create_product
- [x]  get_product
- [x]  list_products
- [x]  delete_product
- [x]  update_product_labels
### productSets.js
- [x]  create_product_set
- [x]  get_product_set
- [x]  list_product_sets
- [x]  delete_product_set
#### –> Products in productSets.js
- [x]  add_product_to_product_set
- [x]  list_products_in_product_set
- [x]  remove_product_from_product_set
### referenceImages.js
- [x]  create_reference_image
- [x]  get_reference_image
- [x]  list_reference_images
- [x]  delete_reference_image
### similarProducts.js
- [x]  get_similar_products_file
- [x]  get_similar_products_uri
### importProductSets.js
- [x] import_product_images
This last sample has been newly added after looking into the python and Java samples for comparison. 

For the canonical versions, please see the [Python Product Search Samples](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/vision/cloud-client/product_search) in [python-docs-samples](https://github.com/GoogleCloudPlatform/python-docs-samples/)

These samples are in Beta and require using the v1p3beta1 API endpoint:
```javascript
const vision = require('@google-cloud/vision').v1p3beta1;
const client = new vision.ProductSearchClient();
```
Before being ready for review & merge:

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary) - working on this one - README is not getting updated with the Product Search code - looking into it.

### CODE REVIEW

**Comment by nirupa-kumar** [comment]
@becca @JustinBeckwith 
Please let me know if this makes more sense. Created a new branch - rebased and including all the Product Search related code. If this seems ok, will go ahead and close the old PR (https://github.com/googleapis/nodejs-vision/pull/192).

File: samples/package.json [hunk 1]
```diff
@@ -6,13 +6,13 @@
   "author": "Google Inc.",
   "repository": "googleapis/nodejs-vision",
   "engines": {
-    "node": ">=8"
+    "node": ">=8.0.0"
```
Comment (JustinBeckwith) [inline samples/package.json]: This change isn't really needed (at least I don't think it is)

File: samples/package.json [hunk 2]
```diff
@@ -6,13 +6,13 @@
...
   },
   "scripts": {
     "test": "ava -T 1m --verbose system-test/*.test.js"
   },
   "dependencies": {
-    "@google-cloud/vision": "^0.22.0",
+    "@google-cloud/vision": "^0.21.0",
```
Comment (JustinBeckwith) [inline samples/package.json]: Please switch this back to 22

File: samples/package.json (hunk 1 above)
Comment (JustinBeckwith) [inline samples/package.json]: Please delete the `samples/package-lock.json` file

File: samples/productSearch/system-test/.eslintrc.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (JustinBeckwith) [inline samples/productSearch/system-test/.eslintrc.yml]: @beccca @kinwa91 I think this is the first place we're putting samples two levels deep like this.  Is this going to work with our kokoro setup?

File: smoke-test/image_annotator_smoke_test.js [hunk 4]
```diff
@@ -18,30 +18,30 @@ describe('ImageAnnotatorSmokeTest', () => {
   it('successfully makes a call to the service', done => {
     const vision = require('../src');
 
-    var client = new vision.v1p3beta1.ImageAnnotatorClient({
+    const client = new vision.v1p3beta1.ImageAnnotatorClient({
```
Comment (JustinBeckwith) [inline smoke-test/image_annotator_smoke_test.js]: This should have been picked up by another PR.  Can you please rebase?

**Comment by becca** [comment]
@nirupa-kumar @JustinBeckwith 👋 hey there! i think you might have pinged the wrong `becca` 😊

File: samples/productSearch/system-test/.eslintrc.yml (hunk 3 above)
Comment (nirupa-kumar) [reply samples/productSearch/system-test/.eslintrc.yml]: Adding the right id:  @beccataylor

File: samples/package.json (hunk 1 above)
Comment (nirupa-kumar) [reply samples/package.json]: Updated this in the latest commit.

**Comment by nirupa-kumar** [comment]
@beccataylor 
I have fixed all the issues reported as part of https://github.com/googleapis/nodejs-vision/pull/192
There is one open issue with the Kokoro tests running , please have a look and let me know if there is anything else required.

File: samples/productSearch/importProductSets.v1p3beta1.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+ * limitations under the License.
+ */
+
+'use strict';
+// [START vision_product_search_import_product_images]
+function importProductSets(projectId, location, gcsUri) {
+  // Imports the Google Cloud client library
+  const vision = require('@google-cloud/vision').v1p3beta1;
+  // Creates a client
+  var client = new vision.ProductSearchClient();
```
Comment (JustinBeckwith) [inline samples/productSearch/importProductSets.v1p3beta1.js]: Please use `const` and `let` everywhere (no var)

**Comment by JustinBeckwith** [comment]
LGTM w/ nits

**Comment by nirupa-kumar** [comment]
@JustinBeckwith 
Fixed the nits. Please approve.

**Comment by JustinBeckwith** [comment]
Thanks!  There are still many instances of `var`.  Would you mind switching these to `let` / `const`?

**Comment by beccasaurus** [comment]
Aaaaaaaannnnnnndddddd.... MERGED 🙌

**Pull request merged**